store.go

```diff
@@ -82,6 +82,14 @@ func (s *Store) RegisterProvider(name string, f Provider) {
 	s.providers[name] = f
 }
 
+// UnregisterProvider unregisters a provider
+func (s *Store) UnregisterProvider(name string) {
+	s.pMut.Lock()
+	defer s.pMut.Unlock()
+	delete(s.providers, name)
```

Comment:
To note: if the provider registered was in conflict with a provider of the same name, and generated a `ProviderConflictErrorLabel` error provider, then this conflict provider won't be unregistered.
I guess for the time being, it's okay.